instrumentation/github.com/aws/aws-sdk-go-v2/otelaws/snsattributes.go

```diff
@@ -0,0 +1,49 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package otelaws // import "go.opentelemetry.io/contrib/instrumentation/github.com/aws/aws-sdk-go-v2/otelaws"
+
+import (
+	"context"
+	"strings"
+
+	"github.com/aws/aws-sdk-go-v2/service/sns"
+	"github.com/aws/smithy-go/middleware"
+
+	"go.opentelemetry.io/otel/attribute"
+	semconv "go.opentelemetry.io/otel/semconv/v1.27.0"
+)
+
+// SNSAttributeSetter sets SNS specific attributes depending on the SNS operation being performed.
+func SNSAttributeSetter(ctx context.Context, in middleware.InitializeInput) []attribute.KeyValue {
+	snsAttributes := []attribute.KeyValue{semconv.MessagingSystemKey.String("aws.sns")}
```

Comment:
This is [`aws_sqs` for SQS](https://github.com/open-telemetry/semantic-conventions/blob/0edb9e2b42e37739fe53e10623d93809269dd300/model/messaging/registry.yaml#L167).

```suggestion
	snsAttributes := []attribute.KeyValue{semconv.MessagingSystemKey.String("aws_sns")}
```